DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleViewControllerBase.cs

```diff
@@ -0,0 +1,299 @@
+{
+    using System;
+    using System.Collections;
+    using System.ComponentModel;
+    using System.IO;
+    using System.Threading;
+    using System.Web.Mvc;
+    using DotNetNuke.Common;
+    using DotNetNuke.Entities.Modules;
+    using DotNetNuke.Entities.Modules.Actions;
+    using DotNetNuke.Entities.Portals;
+    using DotNetNuke.Entities.Users;
+    using DotNetNuke.Internal.SourceGenerators;
+    using DotNetNuke.Services.Localization;
+    using DotNetNuke.UI.Modules;
+    using DotNetNuke.Web.MvcPipeline.Models;
+    using DotNetNuke.Web.MvcPipeline.UI.Utilities;
+
+    public abstract class ModuleViewControllerBase : Controller, IMvcController
+    {
+        /*
+        private ModuleInfo activeModule;
+        private readonly ILog tracelLogger = LoggerSource.Instance.GetLogger("DNN.Trace");
+        private readonly Lazy<ServiceScopeContainer> serviceScopeContainer = new Lazy<ServiceScopeContainer>(ServiceScopeContainer.GetRequestOrCreateScope);
+        private string localResourceFile;
+        private ModuleInstanceContext moduleContext;
+        private DesktopModuleInfo desktopModule;
+        */
+
+        private string localResourceFile;
+        private ModuleInstanceContext moduleContext;
+
+        public ModuleViewControllerBase()
+        {
+        }
+
+        public bool IsHostMenu
+        {
+            get
+            {
+                return Globals.IsHostTab(this.PortalSettings.ActiveTab.TabID);
+            }
+        }
+
+        public PortalSettings PortalSettings
+        {
+            get
+            {
+                return PortalController.Instance.GetCurrentPortalSettings();
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the EditMode property is used to determine whether the user is in the
+        /// Administrator role
+        /// Cache.
+        /// </summary>
+        public bool EditMode
+        {
+            get
+            {
+                return this.ModuleContext.EditMode;
+            }
+        }
+
+        public bool IsEditable
+        {
+            get
+            {
+                return this.ModuleContext.IsEditable;
+            }
+        }
+
+        public int PortalId
+        {
+            get
+            {
+                return this.ModuleContext.PortalId;
+            }
+        }
+
+        public int TabId
+        {
+            get
+            {
+                return this.ModuleContext.TabId;
+            }
+        }
+
+        public UserInfo UserInfo
+        {
+            get
+            {
+                return this.PortalSettings.UserInfo;
+            }
+        }
+
+        public int UserId
+        {
+            get
+            {
+                return this.PortalSettings.UserId;
+            }
+        }
+
+        public PortalAliasInfo PortalAlias
+        {
+            get
+            {
+                return this.PortalSettings.PortalAlias;
+            }
+        }
```

Comment:
Same for PortalSetings which is already exposed, do we need those properties also exposed?